tiledb/sm/filter/filter.cc

```diff
@@ -30,70 +30,38 @@
  * This file defines class Filter.
  */
 
-#include "tiledb/sm/filter/filter.h"
+#include "filter.h"
+#include "bit_width_reduction_filter.h"
```

Comment:
Do we really need all those headers included now that you moved out Filter::create?